pkg/action/sysext.go

```diff
@@ -0,0 +1,356 @@
+package action
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/distribution/reference"
+	"github.com/kairos-io/kairos-agent/v2/pkg/config"
+	"github.com/kairos-io/kairos-sdk/types"
+	"github.com/twpayne/go-vfs/v5"
+)
+
+// Implementation details for not trusted boot
+// sysext are stored under
+// /var/lib/kairos/extensions/
+// we link them to /var/lib/kairos/extensions/{active,passive} depending on where we want it to be enabled
+// Immucore on boot after mounting the persistent dir, will check those dirs\
+// it will then create the proper links to them under /run/extensions
+// This means they are enabled on boot and they are ephemeral, nothing is left behind in the actual sysext dirs
+// This prevents us from having to clean up in different dirs, we can just do cleaning in our dirs (remove links)
+// and on reboot they will not be enabled on boot
+// So all the actions (list, upgrade, download, remove) will be done on the persistent dir
+// And on boot we dinamycally link and enable them based on the boot type (active,passive) via immucore
+
+// Trusted boot works very differently
+// In there, we copy the sysext into the EFI dir under the entry /EFI/kairos/$EFI_FILE.efi.extra.d/EXTENSION_FILE
+// That makes it so systemd-boot will see it, measure it and pass it to the initramfs under the /.extra/sysext dir
+// That means two things
+// We cannot soft link it as EFI partition is FAT
+// To add/remove/update we need to deal with the actual files
+// Currently we are NOT using the measuring of the sysextensions so maybe we could move into a unified system with the same behaviour as above?
+// The sysextensions in Trusted boot are SIGNED so the measurement is not that important and immucore will refuse to move
+// them into /run/extensions if the signature doesnt match, so the measurements are not that important currently and
+// maybe they are not useful if sysext is the way of upgrading your system
+
+// So potentially we have a unified system where we have the same behaviour for both, but that measn that
+// anything provided by sysext will NOT be available until the initramfs stage where we enable the sysext service
+// we can potentially move the sysext service as early as possible once we mount the persistent dir
+
+// Extensions are provided in the following formats:
+// - .raw : image based extension
+// - any directory : directory based extension
+
+// TODO: Check which extensions are running
+// TODO: On disable we should check if the extension is running and refresh systemd-sysext?
+// TODO: On enable, should we refresh systemd-sysext immediately? Only if the current boot state equals the bootstate we enabled it for?
+
+const (
+	sysextDir        = "/var/lib/kairos/extensions/"
+	sysextDirActive  = "/var/lib/kairos/extensions/active"
+	sysextDirPassive = "/var/lib/kairos/extensions/passive"
+	sysextEnabledDir = "/run/extensions/"
```

Comment:
where is this used? I can't see it used anywhere in the changeset

Reply:
this is where immucore will copy the extensions on boot. I wondered if we were gonna use it in the future or not and had it here as a reference. Will remove as it seems we will not use it from here